<?xml version="1.0"?>
<launch>
  <arg name="debug" default="False"/>
  <arg name="use_ros_control" default="True"/>
  <arg name="rt_frequency" default="50.0"/>
  <arg name="ctrl_loop_frequency" default="100.0"/>
  <arg name="min_payload"  default="0.0"/>
  <arg name="max_payload"  default="5.0"/>
  <arg name="prefix" default="" />
  <!-- The max_velocity parameter is only used for debugging in the qr_driver. It's not related to actual velocity limits -->
  <arg name="max_velocity" default="10.0"/>
  <!-- Load the URDF/Xacro model of our robot -->
  <param name="robot_description" command="$(find xacro)/xacro.py '$(find dragon_description)/urdf/single_leg.urdf.xacro'" />

  <!-- Load hardware interface -->
  <node name="qr_driver" pkg="dragon_driver" type="dragon_driver" output="screen">
    <param name="robot_xml_string" command="cat '$(find dragon_driver)/config/robot_specification.xml'"/>
    <param name="use_ros_control" type="bool" value="$(arg use_ros_control)"/>
    <param name="debug" type="bool" value="$(arg debug)" />
    <param name="rt_frequency" type="double" value="$(arg rt_frequency)" />
    <param name="ctrl_loop_frequency" type="double" value="$(arg ctrl_loop_frequency)" />
    <param name="min_payload" type="double" value="$(arg min_payload)"/>
    <param name="max_payload" type="double" value="$(arg max_payload)"/>
    <param name="max_velocity" type="double" value="$(arg max_velocity)"/>
    <param name="prefix" value="$(arg prefix)" />
  </node>

  <!-- Load controller settings -->
  <rosparam if="$(arg use_ros_control)" file="$(find dragon_driver)/config/controllers.yaml" command="load"/>

  <!-- spawn controller manager -->
  <node if="$(arg use_ros_control)" name="ros_control_controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" args="single_leg_velocity_controller" />

  <!-- load other controller --> 
  <!--node name="ros_control_controller_manager" pkg="controller_manager" type="controller_manager" respawn="false"
    output="screen" args="load pos_based_pos_traj_controller" /-->

  <!-- Convert joint states to /tf tranforms -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

</launch>

